### PR TITLE
Bug 1873043: Commit test/e2e/generated/bindata.go

### PR DESCRIPTION
**Pull request by marun**
Upstream ignores this file but has committed it which is somewhat contradictory. Since origin expects to be able to use this path via vendoring, it's important that o/k ensure that an up-to-date version of this file is committed.

This PR needs to merge before origin can be bumped to kube 1.19.0 and beyond.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -120,8 +120,8 @@ zz_generated_*_test.go
 # make-related metadata
 /.make/
 
-# Just in time generated data in the source, should never be committed
-/test/e2e/generated/bindata.go
+# Need to commit so it can be vendored into origin
```
Comment (smarterclayton) [inline .gitignore:123]: You can put this inside the test/e2e/generated dir as:

```
!bindata.go
```

and that will prevent you having to merge this file if the contents change.

File: hack/make-rules/clean.sh [hunk 2]
```diff
@@ -25,7 +25,6 @@ CLEAN_PATTERNS=(
   "_tmp"
   "doc_tmp"
   "((?!staging\/src\/k8s\.io\/apiextensions-apiserver\/pkg\/generated\/openapi).)*/zz_generated.openapi.go"
-  "test/e2e/generated/bindata.go"
```
Comment (smarterclayton) [inline hack/make-rules/clean.sh:28]: Is this necessary?  Having clean purge that isn't the end of the world as long as regen is correct.

File: .gitignore (hunk 1 above)
Comment (marun) [reply .gitignore:123]: Done

File: hack/make-rules/clean.sh (hunk 2 above)
Comment (marun) [reply hack/make-rules/clean.sh:28]: I think this is desirable. Having `make clean` remove a committed file is what I would describe as surprising.

**Comment by sttts** [comment]
Do we verify the bindata file?

**Comment by sttts** [comment]
We do.

/lgtm
/approve

**Pull request merged**